packet.go

```diff
@@ -73,6 +78,9 @@ func (m *Macaroon) parsePacket(start int) (packet, error) {
 	if i <= 0 {
 		return packet{}, fmt.Errorf("cannot parse field name")
 	}
+	if data[len(data)-1] != '\n' {
```

Comment:
No. There's a check at the top of this function that len(data) >= 6, and it's sliced from 4 later, so the smallest length it can have here is 2.